### PR TITLE
feat: rust implementation for some of the Cairo types

### PR DESCRIPTION
**Pull request by ctoyan**
## Description

Implements [#77](https://github.com/eigerco/giza-axelar-starknet/issues/77) in the `giza-axelar-starknet` repository.
Adds Rust representations for some Cairo types.

## Todos

- [x] Unit tests
- [x] Manual tests
- [x] Documentation
- [x] Connect epics/issues

## Steps to Test
Run the unit tests.

### CODE REVIEW

File: ampd/src/starknet/types/array_span.rs [hunk 1]
```diff
@@ -0,0 +1,293 @@
+use starknet_core::types::{FieldElement, ValueOutOfRangeError};
+use thiserror::Error;
+
+/// Represents Cairo's Array and Span types.
+/// Implements `TryFrom<Vec<FieldElement>>`, which is the way to create it.
+///
+/// ## Example usage with the strging "hello"
```
Comment (tomekpiotrowski) [inline ampd/src/starknet/types/array_span.rs:7]: typo

File: ampd/src/starknet/types/array_span.rs [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            Ok(b) => b,
+            Err(e) => return Err(e),
+        };
+
+        Ok(ArraySpan { bytes })
+    }
+}
+
+#[cfg(test)]
+mod array_span_tests {
```
Comment (tomekpiotrowski) [inline ampd/src/starknet/types/array_span.rs:108]: :+1:

File: ampd/src/starknet/types/byte_array.rs [hunk 3]
```diff
@@ -0,0 +1,551 @@
+use itertools::FoldWhile::{Continue, Done};
+use itertools::Itertools;
+use starknet_core::types::{FieldElement, ValueOutOfRangeError};
+use starknet_core::utils::parse_cairo_short_string;
+use thiserror::Error;
+
+/// Represents Cairo's ByteArray type.
+/// Implements `TryFrom<Vec<FieldElement>>`, which is the way to create it.
+///
+/// ## Example usage with the strging "hello"
```
Comment (tomekpiotrowski) [inline ampd/src/starknet/types/byte_array.rs:10]: same typo

**Review by tomekpiotrowski — CHANGES_REQUESTED** [review]

**Review by tomekpiotrowski — APPROVED** [review]

**Comment by ctoyan** [comment]
Test pass and clippy passes locally.
I presume that the CI doesn't pass, because Axelar use a custom workflow runner and in our organisation the tasks just stay queued and are never picked up.

**Pull request merged**